rectools/models/sasrec.py

```diff
@@ -78,24 +153,85 @@ def __init__(self, n_factors: int, n_items: int, dropout_rate: float):
 
     def forward(self, items: torch.Tensor) -> torch.Tensor:
         """TODO"""
-        item_embs = self.item_emb(items)
+        item_embs = self.ids_emb(items)
         item_embs = self.drop_layer(item_embs)
         return item_embs
 
+    @property
+    def device(self) -> torch.device:
+        """TODO"""
+        return self.ids_emb.weight.device
+
+    @classmethod
+    def from_dataset(cls, dataset: Dataset, n_factors: int, dropout_rate: float) -> tpe.Self:
+        """TODO"""
+        n_items = dataset.item_id_map.size
+        return cls(n_factors, n_items, dropout_rate)
+
+
+class ItemNetConstructor(ItemNetBase):
+    """
+    Base class constructor for ItemNet, taking as input a sequence of ItemNetBase nets,
+    including custom ItemNetBase nets.
+    Constructs item's embedding based on aggregation of its embeddings from the passed networks.
+    """
+
+    def __init__(
+        self,
+        n_items: int,
+        item_net_blocks: tp.Sequence[ItemNetBase],
+    ) -> None:
+        """TODO"""
+        super().__init__()
+
+        if len(item_net_blocks) == 0:
+            raise ValueError("At least one type of net to calculate item embeddings should be provided.")
+
+        self.n_items = n_items
+        self.n_item_blocks = len(item_net_blocks)
+        self.item_net_blocks = nn.ModuleList(item_net_blocks)
+
+    def forward(self, items: torch.Tensor) -> torch.Tensor:
+        """TODO"""
+        item_embs = []
+        # TODO: Add functionality for parallel computing.
+        for idx_block in range(self.n_item_blocks):
+            item_emb = self.item_net_blocks[idx_block](items)
+            item_embs.append(item_emb)
+        return torch.sum(torch.stack(item_embs, dim=0), dim=0)
+
+    @property
+    def device(self) -> torch.device:
+        """TODO"""
+        device = self.item_net_blocks[0].device
+        return device
+
     @property
     def catalogue(self) -> torch.Tensor:
         """TODO"""
-        return torch.arange(0, self.n_items, device=self.item_emb.weight.device)
+        return torch.arange(0, self.n_items, device=self.device)
 
     def get_all_embeddings(self) -> torch.Tensor:
         """TODO"""
         return self.forward(self.catalogue)
 
     @classmethod
-    def from_dataset(cls, dataset: Dataset, n_factors: int, dropout_rate: float) -> tpe.Self:
+    def construct_nets_from_dataset(
```

Comment:
I kind of like that you inherit `ItemNetConstructor` from `ItemNetBase`. Let's keep it up.
Let's rename `construct_nets_from_dataset` method to `from_dataset` and treat all of the following logic as a usual `ItemNetBase` `from_dataset` behaviour. It makes sense.